apps/plea/models.py

```diff
@@ -452,30 +452,58 @@ def has_court(self, urn):
         """
         Take a URN and return True if the region_code is valid
         """
+        return self.filter(region_code=urn[:2],
+                           enabled=True).exists()
 
-        try:
-            self.get(region_code=urn[:2],
-                     enabled=True)
+    def get_by_urn(self, urn):
+        """
+        Retrieve court model by URN
+        """
 
-            return True
+        courts = self.filter(region_code=urn[:2],
+                             enabled=True).order_by("id")
 
-        except Court.DoesNotExist:
-            return False
+        if courts:
+            return courts[0]
+        else:
+            raise Court.DoesNotExist
 
-    def get_by_urn(self, urn):
+    def get_court(self, urn, ou_code=None):
         """
-        Retrieve court model by URN
+        Attempt to return the court by URN or ou code.
+
+        Prioritise matching on ou code but if there is no match then attempt to match on URN 
         """
 
-        return self.get(region_code=urn[:2],
-                        enabled=True)
+        if ou_code:
+            try:
+                return OUCode.objects.get(ou_code=ou_code[:5]).court
+            except OUCode.DoesNotExist:
+                pass
+
+        return self.get_by_urn(urn)
+
+    def get_court_by_ou_code(self, ou_code):
+        return self.select_related().get(oucode__ou_code=ou_code[:4])
+
+    def get_court_dx(self, urn):
+        """
+        Get court whilst using DX data and ou-code matching where possible
+        """
+
+        try:
+            ou_code = Case.objects.get(urn=standardise_urn(urn), imported=True, ou_code__isnull=False).ou_code
```

Comment:
on our projects, we tend to stick to 79 characters per line. sometimes it does create weird looking lines trying to fight we the limit.